lib/archethic/contracts/interpreter/function_interpreter.ex

```diff
@@ -98,18 +97,78 @@ defmodule Archethic.Contracts.Interpreter.FunctionInterpreter do
   #  | .__/|_|  \___| \_/\_/ \__,_|_|_|\_\
   #  |_|
   # ----------------------------------------------------------------------
-
   # Ban access to Contract module
   defp prewalk(
-         node = {:__aliases__, _, [atom: "Contract"]},
-         _
+         node =
+           {{:., _meta, [{:__aliases__, _, [atom: "Contract"]}, {:atom, function_name}]}, _, _},
+         acc,
+         _visibility
+       ) do
+    absolute_module_atom =
+      Code.ensure_loaded!(
+        String.to_existing_atom("Elixir.Archethic.Contracts.Interpreter.Library.Contract")
```

Comment:
Since this module will soon support function that do not write the contract, it could goes in Common module